scripts/cluster/join.py

```diff
@@ -36,6 +38,23 @@
 cluster_key_file = "{}/cluster.key".format(cluster_dir)
 
 
+def get_fingerprint(addr, port):
+    """
+    Get the certificate fingerprint of the server at addr:port
+
+    :param addr: the address of the server
+    :param port: the port of the server
+    :return: the md5sum of the server cert
+    """
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.settimeout(1)
+    wrapped_socket = ssl.wrap_socket(sock)
+    wrapped_socket.connect((addr, port))
+    der_cert_bin = wrapped_socket.getpeercert(True)
+    wrapped_socket.close()
+    return hashlib.md5(der_cert_bin).hexdigest()
```

Comment:
As discussed online, MD5 is insecure.  And SHA256 is only 65 chars vs MD5's 33, so it definitely seems worth switching to SHA256.

Reply:
+1 doing it now. Thank you